application/F3DStarter.cxx

```diff
@@ -653,7 +686,9 @@ int F3DStarter::AddFile(const fs::path& path, bool quiet)
 
     if (it == this->Internals->FilesList.end())
     {
+      this->Internals->FilesListMutex.lock();
       this->Internals->FilesList.push_back(tmpPath);
+      this->Internals->FilesListMutex.unlock();
       return static_cast<int>(this->Internals->FilesList.size()) - 1;
```

Comment:
get size before unlocking
```suggestion
      this->Internals->FilesList.push_back(tmpPath);
      const size_t fileCount = this->Internals->FilesList.size();
      this->Internals->FilesListMutex.unlock();
      return static_cast<int>(fileCount) - 1;
```

Reply:
line 690 is the only place where FilesList can be modfied. Concurrent read doesnt matter, does it ?

Reply:
I don't know, it _feels_ like if you're locking around the push you may want to get the updated size before another push has a chance to get in there.
But I you know for sure that specific piece of code isn't going to be concurrent with itself I guess it's fine either way

Reply:
yes, this mutex is there ONLY for the dmon case. There are many other read of FileLists everywhere in the file, I'd rather not add mutex code there either. I can add a comment though.

Reply:
Let's use a `std::lock_guard` anyway